core/src/main/java/io/questdb/std/Numbers.java

```diff
@@ -973,6 +974,94 @@ public static long parseLongSize(CharSequence sequence) throws NumericException
         return negative ? val : -val;
     }
 
+    public static long parseLongDuration(CharSequence sequence) throws NumericException {
```

Comment:
it would be good to have a comprehensive unit test of this function separately to testing file purging logic. Also I feel it could be simplified, but it would be easier to do after there is a test for it